xbmc/Application.cpp

```diff
@@ -2692,9 +2692,13 @@ bool CApplication::OnMessage(CGUIMessage& message)
                                                          m_itemCurrentFile, param);
 
       // we don't want a busy dialog when switching channels
-      if (!m_itemCurrentFile->IsLiveTV())
+      const int activeWindowID = CServiceBroker::GetGUI()->GetWindowManager().GetActiveWindow();
+      if (!m_itemCurrentFile->IsLiveTV() ||
```

Comment:
wouldn't it be possible to check `|| (m_itemCurrentFile->IsLiveTV() && !m_appPlayer.IsPlaying())`  instead? If it works it could be a possible way to avoid the GUI dependencies here

Reply:
Just tested. Does not work. `m_appPlayer.IsPlaying()` seems already to return true when we do the check, even if we tune the first channel and nothing else is playing.

Reply:
Hum, maybe `IsPlayingVideo()` for tv and `IsPlayingAudio()` for radio will work? I guess those should come way later than player.playing.

Reply:
That seems to work. :-) Will change the code. Thanks.

Reply:
It has a small side effect though.
If you are playing music in the background of the homescreen and you start a channel, you'll not show a spinner.
So again first channel start will result in no user feedback?

Reply:
@a1rwulf oops, not tested, but you may be right. So, it is not a perfect solution, but an improvement. Important is not to show the spinner when switching channels. If there is another case where we should display the spinner, this could be done in another PR (as I have no idea how to solve this - and I'm with @enen92 regarding GUI dependency - active window check - not being optimal).